### PR TITLE
Wip kr template

### PR DESCRIPTION
**Pull request by zoperado**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Changelog updated
- [ ] Readme updated
- [ ] Tests updated

### CODE REVIEW

File: hcs-sxc-java-core/src/main/java/com/hedera/hcs/sxc/consensus/OutboundHCSMessage.java [hunk 1]
```diff
@@ -195,43 +237,119 @@ public TransactionId sendMessage(int topicIndex, byte[] message) throws Exceptio
...
                 log.info("Executing transaction");
                 TransactionId txId = tx.execute(client);
-
+                
                 TransactionReceipt receipt = txId.getReceipt(client, Duration.ofSeconds(30));
 
                 transactionId = new TransactionId(this.operatorAccountId);
 
-                log.info("status is {} "
+                log.info("Mesage receipt status is {} "
```
Comment (gregscullard) [inline hcs-sxc-java-core/src/main/java/com/hedera/hcs/sxc/consensus/OutboundHCSMessage.java]: Typo - Message

File: hcs-sxc-java-examples/hcs-sxc-java-settlement-demo/config/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
+version: '3.3'
```
Comment (gregscullard) [inline hcs-sxc-java-examples/hcs-sxc-java-settlement-demo/config/docker-compose.yml]: docker-compose shouldn't really be checked in. Should be .gitignored

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/pom.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/pom.xml]: Should rename plugin to indicate which type of encryption we use here to allow for different ones in the future

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java [hunk 4]
```diff
@@ -30,41 +30,44 @@
...
     @BeforeAll
-    public static void testInitClass(){
-        KeyRotation keyRotation = new KeyRotation();
-        byte[] alicePublic = keyRotation.aliceFirst();
-        Pair<byte[], byte[]> bobPubSecret = KeyRotation.bobGenFromAlice(alicePublic);
-        byte[] aliceSharedSecret = keyRotation.aliceFinish(bobPubSecret.getLeft());
-        byte[] bobSharedSecret = bobPubSecret.getRight();
-        assertTrue(Arrays.equals(aliceSharedSecret, bobSharedSecret));
-        sharedSecret = aliceSharedSecret;
+    public static void initClass(){
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java]: Prefix method name with `test` ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java [hunk 5]
```diff
@@ -30,41 +30,44 @@
...
+        } catch (Exception ex) {
+            Logger.getLogger(CryptographyTest.class.getName()).log(Level.SEVERE, null, ex);
+        }
     }
     
-    
     @Test
-    public void testEncryptAndDecrypt() throws Exception {
-        byte[] encrypt = Cryptography.encrypt(sharedSecret, cleartext);
+    public void encryptAndDecrypt() throws Exception {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java]: prefix method name with `test` ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java [hunk 6]
```diff
@@ -454,4 +456,50 @@ public void clear() {
         session.createQuery("delete HCSApplicationMessage").executeUpdate();
         session.getTransaction().commit();
     }
+
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java]: Should this be in the `PersistMessages` class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java [hunk 7]
```diff
@@ -454,4 +456,50 @@ public void clear() {
...
+            KeyStore ks = new KeyStore();
+            ks.setId(0);
+            ks.setSecretKey(secretKey);
+            session.save(ks);
+        }
+        
+        
+    }
+
+    @Override
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java [hunk 8]
```diff
@@ -454,4 +456,50 @@ public void clear() {
...
+    public byte[] getSecretKey() {
+        final Session session = HibernateUtil.getHibernateSession(this.hibernateProperties);
+        KeyStore ks = session.find(KeyStore.class, 0);
+        return ks.getSecretKey();
+    }   
+
+    @Override
+    public void storePublicKey(byte[] publicKey) {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java [hunk 9]
```diff
@@ -454,4 +456,50 @@ public void clear() {
...
+        final Session session = HibernateUtil.getHibernateSession(this.hibernateProperties);
+        KeyStore ks = session.find(KeyStore.class, 0);
+        ks.setPublicKey(publicKey);
+        session.save(ks);
+        
+    }
+
+    @Override
+    public byte[] getPublicKey() {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/src/main/java/com/hedera/hcs/sxc/plugin/persistence/hibernate/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java [hunk 10]
```diff
@@ -200,4 +201,26 @@ public void clear() {
     public void setHibernateProperties(Map<String, String> hibernateProperties) {
         
     }
+
+    @Override
+    public void storeSecretKey(byte[] secretKey) {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java [hunk 11]
```diff
@@ -200,4 +201,26 @@ public void clear() {
...
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    @Override
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java [hunk 12]
```diff
@@ -200,4 +201,26 @@ public void clear() {
...
+    public byte[] getSecretKey() {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    @Override
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java]: Should this be in the PersistMessages class ?

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java [hunk 13]
```diff
@@ -200,4 +201,26 @@ public void clear() {
...
+    public void storePublicKey(byte[] secretKey) {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    @Override
+    public byte[] getPublicKey() {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/src/main/java/com/hedera/hcs/sxc/plugin/persistence/inmemory/PersistMessages.java]: Should this be in the PersistMessages class ?

File: pom.xml [hunk 14]
```diff
@@ -186,14 +186,12 @@
   <build>
     <pluginManagement>
        <plugins>
+         <!--
```
Comment (gregscullard) [inline pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-core/pom.xml [hunk 15]
```diff
@@ -41,12 +42,12 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-core/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-interfaces/pom.xml [hunk 16]
```diff
@@ -23,12 +23,13 @@
             <version>${hcs-sxc.version}</version>
         </dependency>
     </dependencies>
-    <build>
+    
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-interfaces/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-direct/pom.xml [hunk 17]
```diff
@@ -24,13 +24,13 @@
             <version>${hcs-sxc.version}</version>
         </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-direct/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-queue-artemis/pom.xml [hunk 18]
```diff
@@ -29,13 +30,14 @@
           <version>${artemis-jms-client-all.version}</version>
       </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-queue-artemis/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/pom.xml [hunk 19]
```diff
@@ -34,14 +34,14 @@
             <version>${hcs-sxc.version}</version>
         </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-hibernate/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/pom.xml [hunk 20]
```diff
@@ -29,13 +29,13 @@
             <version>${hcs-sxc.version}</version>
         </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-persistence-in-memory/pom.xml]: should re-enable through profile at a minimum

File: hcs-sxc-java-relay/pom.xml [hunk 21]
```diff
@@ -28,14 +28,14 @@
           <version>${artemis-jms-client-all.version}</version>
       </dependency>
     </dependencies>
-    <build>
+<!--    <build>
```
Comment (gregscullard) [inline hcs-sxc-java-relay/pom.xml]: should re-enable through profile at a minimum

File: pom.xml [hunk 22]
```diff
@@ -38,7 +38,7 @@
       <module>hcs-sxc-java-examples</module>
       <module>hcs-sxc-java-proto</module>
       <module>hcs-sxc-java-plugins</module>
-      <module>hcs-sxc-java-coverage</module>
+      <!--<module>hcs-sxc-java-coverage</module>-->
```
Comment (gregscullard) [inline pom.xml]: should re-enable

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java [hunk 23]
```diff
@@ -1,143 +1,159 @@
...
+import javax.crypto.interfaces.DHPublicKey;
+import javax.crypto.spec.DHParameterSpec;
+import org.apache.commons.lang3.tuple.Pair;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class KeyRotation implements SxcKeyRotation{
+
+    private KeyAgreement aliceKeyAgree;
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java]: Replace `alice` with `initiator`

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java [hunk 24]
```diff
@@ -1,143 +1,159 @@
...
+     * for Bob's generated public key, obtained using
+     * {@link #bobGenFromAlice(byte[] alicesPublicKey)},  
+     * so that she can  generate the shared key herself by 
+     * executing {@link #aliceFinish(byte[] bobsPublicKey)} 
+     * Alice needs to reuse the KeyAgreement when finalizing the exchange
+     *, which is kept in this object's memory, 
+     * 
+     * @return Alice's public key.
+     */
+    private byte[]  aliceFirst() {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java]: Replace alice with initiator

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java [hunk 25]
```diff
@@ -1,143 +1,159 @@
...
+    /**
+     * Uses Alice's public key generated by {@link #aliceFirst()} and obtains Bob's 
+     * public key and shared secret. The public key is meant to be given to Alice
+     * so that she can construct the same shared key herself. 
+     * Note that this is a static method, Bob does not need to construct the object. 
+     * 
+     * @param alicePubKeyEnc
+     * @return Pair.of(bob's public key, shared secret)
+     */
+    private  Pair<byte[],byte[]> bobGenFromAlice(byte[] alicePubKeyEnc) {
```
Comment (gregscullard) [inline hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java]: Replace `bob` with `responder`

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/pom.xml (hunk 3 above)
Comment (zoperado) [reply hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/pom.xml]: the cryptography plugin contains both, encryption and keyrotation packages.  Best at this stage to rename at the package level.

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java (hunk 4 above)
Comment (zoperado) [reply hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java]: done

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java (hunk 5 above)
Comment (zoperado) [reply hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/test/java/com/hedera/hcs/sxc/plugin/cryptography/cryptography/CryptographyTest.java]: done

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java (hunk 23 above)
Comment (zoperado) [reply hcs-sxc-java-plugins/hcs-sxc-java-plugins-cryptography/src/main/java/com/hedera/hcs/sxc/plugin/cryptography/keyrotation/KeyRotation.java]: done

File: hcs-sxc-java-relay/pom.xml (hunk 21 above)
Comment (zoperado) [reply hcs-sxc-java-relay/pom.xml]: re enablbed

File: hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-queue-artemis/pom.xml (hunk 18 above)
Comment (zoperado) [reply hcs-sxc-java-plugins/hcs-sxc-java-plugins-mirror-queue-artemis/pom.xml]: done

File: hcs-sxc-java-examples/hcs-sxc-java-settlement-demo/config/docker-compose.yml (hunk 2 above)
Comment (zoperado) [reply hcs-sxc-java-examples/hcs-sxc-java-settlement-demo/config/docker-compose.yml]: done, deleted and kept local

File: hcs-sxc-java-core/src/main/java/com/hedera/hcs/sxc/consensus/OutboundHCSMessage.java (hunk 1 above)
Comment (zoperado) [reply hcs-sxc-java-core/src/main/java/com/hedera/hcs/sxc/consensus/OutboundHCSMessage.java]: done in one more place

**Pull request closed without merging**